tests/network/link_state.go

```diff
@@ -0,0 +1,229 @@
+				bridgeInterfaceName2: v1.InterfaceStateLinkDown,
+			})
+			Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+			var timeout = time.Second * 5
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac1, regexLinkStateUP), timeout)).To(Succeed())
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac2, regexLinkStateDOWN), timeout)).To(Succeed())
+
+			By("flipping the state of both interfaces")
+
+			vm, err := kubevirt.Client().VirtualMachine(vmName.Namespace).Get(context.Background(), vmName.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(PatchFlipVMInterfacesStates(vm)).To(Succeed())
+
+			vmi = WaitForStatusToShowExpectedStates(vmName, map[string]v1.InterfaceState{
+				defaultInterfaceName: v1.InterfaceStateLinkDown,
+				bridgeInterfaceName2: v1.InterfaceStateLinkUp,
+			})
+			timeout = time.Second * 30
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac1, regexLinkStateDOWN), timeout)).To(Succeed())
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac2, regexLinkStateUP), timeout)).To(Succeed())
+		})
+
+		It("hot plugging an interface with link state down and migrating vmi", func() {
+			hotplugNet := newMultusNetwork(bridgeInterfaceName3, bridgeInterfaceNAD3)
+			hotplugIface := libvmi.InterfaceWithMac(libvmi.InterfaceWithState(
+				pointer.P(libvmi.InterfaceDeviceWithBridgeBinding(bridgeInterfaceName3)),
+				v1.InterfaceStateLinkDown),
+				mac3.String())
+
+			vm, err := kubevirt.Client().VirtualMachine(testsuite.GetTestNamespace(nil)).Get(context.Background(), vmName.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(libnet.PatchVMWithNewInterface(vm, *hotplugNet, *hotplugIface)).To(Succeed())
+
+			migration := libmigration.New(vmName.Name, vmName.Namespace)
+			migrationUID := libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+			libmigration.ConfirmVMIPostMigration(kubevirt.Client(), bareNamedVMI(vmName), migrationUID)
+
+			vmi := WaitForStatusToShowExpectedStates(vmName, map[string]v1.InterfaceState{
+				defaultInterfaceName: v1.InterfaceStateLinkUp,
+				bridgeInterfaceName2: v1.InterfaceStateLinkDown,
+				bridgeInterfaceName3: v1.InterfaceStateLinkDown,
+			})
+			Expect(console.LoginToFedora(vmi)).To(Succeed())
+			timeout := time.Second * 30
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac1, regexLinkStateUP), timeout)).To(Succeed())
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac2, regexLinkStateDOWN), timeout)).To(Succeed())
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac3, regexLinkStateDOWN), timeout)).To(Succeed())
+
+		})
+	})
+})
+
+func bareNamedVMI(vmName types.NamespacedName) *v1.VirtualMachineInstance {
+	return &v1.VirtualMachineInstance{
+		ObjectMeta: metav1.ObjectMeta{Name: vmName.Name, Namespace: vmName.Namespace},
+	}
+}
+
+func newMultusNetwork(name, netAttachDefName string) *v1.Network {
```

Comment:
This functionality exists in libvmi:
https://github.com/kubevirt/kubevirt/blob/bfc9404dee4533c4fe621ca95042cf891eff78ef/pkg/libvmi/network.go#L111